hydra/_internal/utils.py

```diff
@@ -516,8 +516,17 @@ def _locate(path: str) -> Union[type, Callable[..., Any]]:
     else:
         obj = builtins
     for part in parts[n:]:
+        mod = mod + "." + part
         if not hasattr(obj, part):
-            raise ImportError(f"Could not locate '{path}'")
+            # Check if there was an import error
+            try:
+                import_module(mod)
+            except Exception as e:
+                raise ImportError(
+                    f"Encountered error: `{e}` when loading module '{path}'"
+                ) from e
+            else:
+                raise ImportError(f"Could not locate '{path}'")
```

Comment:
Why is this needed? the stink is probably at callsite.

Reply:
`hasattr(obj, part)` is False if the imported file (ie `part`) has any errors (including importing an uninstalled module or syntax error).

Reply:
Why do you need to intercept the exception here instead of in the callsite?

Reply:
You mean here: https://github.com/facebookresearch/hydra/blob/master/hydra/utils.py#L32 ?

Reply:
yes. It's already catching exceptions. Can it use the message from the caught exception?

Reply:
Agreed that it is catching exceptions. But I still need `import_module(mod)` to generate the exception. So I will just generate the exception here, and pass it up all the way.